File: 백준/Bronze/25304. 영수증/영수증.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.StringTokenizer;

public class Main {

	public static void main(String[] args) throws IOException {
		
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
		
		int x = Integer.parseInt(br.readLine());
		int y = Integer.parseInt(br.readLine());
		int sum = 0;
		
		for(int i = 1; i <= y; i++) {
			StringTokenizer st = new StringTokenizer(br.readLine());
			int a = Integer.parseInt(st.nextToken());
			int b = Integer.parseInt(st.nextToken());
			
			sum += (a * b);
		}
		
		if(x == sum) {
			bw.write("Yes\n");
		} else {
			bw.write("No\n");
		}
		
		bw.flush();
		bw.close();
		br.close();
		
		
	}
		
}

File: 백준/Bronze/25304. 영수증/영수증.cc
#include <iostream>

using namespace std;

int main()
{
	int price, n;
	cin >> price;
	cin >> n;

	int sum = 0;
	for (int i = 0; i < n; i++) {
		int a, b;
		cin >> a >> b;

		sum += a * b;
	}

	if (price == sum)
		cout << "Yes";
	else
		cout << "No";

	return 0;
}